src/Type/Transaction.php

```diff
@@ -185,38 +185,38 @@ public function __toString(): string
     {
         // TODO: is the padding even necessary?
         $valueTrits = TritsUtil::fromInt($this->value->getAmount(), 81);
-        while (\count($valueTrits) < 81) {
-            $valueTrits[] = 0;
+        if (\count($valueTrits) < 81) {
+            array_merge($valueTrits, array_fill(count($valueTrits), 81 - count($valueTrits), 0));
         }
 
-        $timestampTrits = TritsUtil::fromInt((string)$this->timestamp, 27);
-        while (\count($timestampTrits) < 27) {
-            $timestampTrits[] = 0;
+        $timestampTrits = TritsUtil::fromInt((string) $this->timestamp, 27);
+        if (\count($timestampTrits) < 27) {
```

Comment:
I had that before, but it was really inefficient (maybe because I made it inefficient.. :-)), so I got back to simple arrays for trits - I think it was especially in the SpongeInterface implementations (Kerl, Curl) with the recursion and array access, but I cannot remember exactly. Feel free to propose a better solution.

See my other comment for the \count issue, we can use array_pad.

Reply:
@ace411, thank you for your reply. @Techworker. At this time, I think it's okay to use the `array_pad` function to do this.

The `collect`  approach I've almost known that, but the important work is: Let this can be passed by the unit test in Travis CI build. This work should tag the `enhancement` and it will do this in the future.